exchange/exchange.go

```diff
@@ -214,32 +214,42 @@ type BidderRequest struct {
 }
 
 func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *DebugLog) (*openrtb2.BidResponse, error) {
-	reject := r.HookExecutor.ExecuteProcessedAuctionStage(r.BidRequestWrapper.BidRequest)
-	if reject != nil {
-		return nil, reject
-	}
-
-	var errs []error
-	// rebuild/resync the request in the request wrapper.
-	if err := r.BidRequestWrapper.RebuildRequest(); err != nil {
+	err := r.HookExecutor.ExecuteProcessedAuctionStage(r.BidRequestWrapper)
+	if err != nil {
 		return nil, err
 	}
-	requestExt, err := extractBidRequestExt(r.BidRequestWrapper.BidRequest)
+
+	requestExt, err := r.BidRequestWrapper.GetRequestExt()
+	requestExtPrebid := requestExt.GetPrebid()
 	if err != nil {
 		return nil, err
 	}
+
 	if !e.server.Empty() {
-		requestExt.Prebid.Server = &openrtb_ext.ExtRequestPrebidServer{ExternalUrl: e.server.ExternalUrl, GvlID: e.server.GvlID, DataCenter: e.server.DataCenter}
+		if requestExtPrebid == nil {
+			requestExtPrebid = &openrtb_ext.ExtRequestPrebid{}
+		}
+		requestExtPrebid.Server = &openrtb_ext.ExtRequestPrebidServer{
+			ExternalUrl: e.server.ExternalUrl,
+			GvlID:       e.server.GvlID,
+			DataCenter:  e.server.DataCenter}
```

Comment:
I realize this is probably outside the scope of this PR but, could `ExtRequestPrebidServer` be used?
```
57   type exchange struct {
58       adapterMap               map[openrtb_ext.BidderName]AdaptedBidder
59       bidderInfo               config.BidderInfos
60       bidderToSyncerKey        map[string]string
61       me                       metrics.MetricsEngine
62       cache                    prebid_cache_client.Client
63       cacheTime                time.Duration
64       gdprPermsBuilder         gdpr.PermissionsBuilder
65       tcf2ConfigBuilder        gdpr.TCF2ConfigBuilder
66       currencyConverter        *currency.RateConverter
67       externalURL              string
68       gdprDefaultValue         gdpr.Signal
69       privacyConfig            config.Privacy
70       categoriesFetcher        stored_requests.CategoryFetcher
71       bidIDGenerator           BidIDGenerator
72       hostSChainNode           *openrtb2.SupplyChainNode
73       adsCertSigner            adscert.Signer
74 -     server                   config.Server
   +     server                   *openrtb_ext.ExtRequestPrebidServer
75       bidValidationEnforcement config.Validations
76       requestSplitter          requestSplitter
77   }
exchange/exchange.go
```
so we could maybe simplify to:
```
228     if !e.server.Empty() {
229         if requestExtPrebid == nil {
230             requestExtPrebid = &openrtb_ext.ExtRequestPrebid{}
231         }
232 -       requestExtPrebid.Server = &openrtb_ext.ExtRequestPrebidServer{
233 -           ExternalUrl: e.server.ExternalUrl,
234 -           GvlID:       e.server.GvlID,
235 -           DataCenter:  e.server.DataCenter}
    +       requestExtPrebid.Server = *e.server
236         requestExt.SetPrebid(requestExtPrebid)
237     }
exchange/exchange.go
```

Reply:
I'm not sure. I prefer to leave that out for now. I'm not yet convinced `config.Server` should be replaced wholesale.